src/hooks/safe-apps/useQueryFilter.ts

```diff
@@ -0,0 +1,57 @@
+import { type Dispatch, type SetStateAction, useEffect } from 'react'
+import { useRouter } from 'next/router'
```

Comment:
```suggestion
import { useSearchParams } from 'next/navigation'
```

Reply:
How do we navigate without the router though?

Reply:
Ah, you also use it to change the route. Nvm then :)